apps/wallet/src/background/account-sources/QredoAccountSource.ts

```diff
@@ -82,16 +83,25 @@ export class QredoAccountSource extends AccountSource<QredoAccountSourceSerializ
 				throw new Error('Qredo account source already exists');
 			}
 		}
-		await (await getDB()).accountSources.put(dataSerialized);
-		await backupDB();
-		accountSourcesEvents.emit('accountSourcesChanged');
-		return new QredoAccountSource(dataSerialized.id);
+		return dataSerialized;
 	}
 
 	static isOfType(serialized: AccountSourceSerialized): serialized is QredoAccountSourceSerialized {
 		return serialized.type === 'qredo';
 	}
 
+	static async save(
+		serialized: QredoAccountSourceSerialized,
+		{ skipBackup = false }: { skipBackup?: boolean } = {},
+	) {
+		await (await Dexie.waitFor(getDB())).accountSources.put(serialized);
```

Comment:
should this be a method on the base class `persist` ?

Reply:
Yeah, I might move it in the future if it makes sense when we add more account sources